frame/nfts/src/features/atomic_swap.rs

```diff
@@ -0,0 +1,152 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::*;
+use frame_support::{
+	pallet_prelude::*,
+	traits::{Currency, ExistenceRequirement::KeepAlive},
+};
+
+impl<T: Config<I>, I: 'static> Pallet<T, I> {
+	pub fn do_create_swap(
+		sender: T::AccountId,
+		collection_id: T::CollectionId,
+		item_id: T::ItemId,
+		desired_collection_id: T::CollectionId,
+		desired_item_id: T::ItemId,
+		maybe_price: Option<ItemPrice<T, I>>,
+		maybe_duration: Option<<T as SystemConfig>::BlockNumber>,
+	) -> DispatchResult {
+		let item = Item::<T, I>::get(&collection_id, &item_id).ok_or(Error::<T, I>::UnknownItem)?;
+		ensure!(item.owner == sender, Error::<T, I>::NoPermission);
+
+		Item::<T, I>::get(&desired_collection_id, &desired_item_id)
+			.ok_or(Error::<T, I>::UnknownItem)?;
+
+		let now = frame_system::Pallet::<T>::block_number();
+		let deadline = maybe_duration.map(|d| d.saturating_add(now));
+
+		PendingSwapOf::<T, I>::insert(
+			&collection_id,
+			&item_id,
+			PendingSwap {
+				desired_collection: desired_collection_id,
+				desired_item: desired_item_id,
+				price: maybe_price,
+				deadline,
+			},
+		);
+
+		Self::deposit_event(Event::SwapCreated {
+			collection: collection_id,
+			item: item_id,
+			desired_collection: desired_collection_id,
+			desired_item: desired_item_id,
+			price: maybe_price,
+			deadline,
+		});
+
+		Ok(())
+	}
+
+	pub fn do_cancel_swap(
+		sender: T::AccountId,
+		collection_id: T::CollectionId,
+		item_id: T::ItemId,
+	) -> DispatchResult {
+		let item = Item::<T, I>::get(&collection_id, &item_id).ok_or(Error::<T, I>::UnknownItem)?;
+		let swap = PendingSwapOf::<T, I>::get(&collection_id, &item_id)
+			.ok_or(Error::<T, I>::UnknownSwap)?;
+
+		let is_past_deadline = if let Some(deadline) = swap.deadline {
+			let now = frame_system::Pallet::<T>::block_number();
+			now > deadline
+		} else {
+			false
+		};
+
+		if !is_past_deadline {
+			ensure!(item.owner == sender, Error::<T, I>::NoPermission);
+		}
+
+		PendingSwapOf::<T, I>::remove(&collection_id, &item_id);
+
+		Self::deposit_event(Event::SwapCancelled {
+			collection: collection_id,
+			item: item_id,
+			desired_collection: swap.desired_collection,
+			desired_item: swap.desired_item,
+			price: swap.price,
+			deadline: swap.deadline,
+		});
+
+		Ok(())
+	}
+
+	pub fn do_claim_swap(
+		sender: T::AccountId,
+		send_collection_id: T::CollectionId,
+		send_item_id: T::ItemId,
+		receive_collection_id: T::CollectionId,
+		receive_item_id: T::ItemId,
+		maybe_receive_amount: Option<ItemPrice<T, I>>,
```

Comment:
sender and receiver could be profferer and acceptor or swap_creator and swap_completer? Both participants send and receive so they're words to be avoided ideally.

Reply:
I think send_item_id/receive_item_id is kinda valid, because from a point of view of the caller it's totally legit.
The receiving end is going to have those values swapped around.

At the same time you could do something like
`swapped_item_id` and `received_item_id` if that's more readable. 

I'd say that whatever makes this more readable and understandable is the best choice.